python/fledge/services/core/server.py

```diff
@@ -1615,3 +1642,107 @@ async def add_audit(cls, request):
             raise web.HTTPInternalServerError(reason=ex)
 
         return web.json_response(message)
+
+    @classmethod
+    async def verify_token(cls, request):
+        """ Endpoint for verifycation of service bearer token received at registration time
+
+        :Example:
+            curl -X POST -d '{"bearer_token" : "hjhdjsfshhj"}
+             http://localhost:<core mngmt port>/fledge/service/verity_token
+        """
+        data = await request.json()
+        bearer_token = data.get('bearer_token', None)
+        if bearer_token is not None:
+            # Check input token exists in system
+            foundToken = False
+            services_list = ServiceRegistry.all()
+            for service in services_list:
+                t = service._bearer_token if service._bearer_token is not None else ""
+                if t == bearer_token:
+                    foundToken = True
+                    break
+            # Raise error if token does not exists
+            if foundToken == False:
+                msg = 'token does not exist in system'
+                raise web.HTTPBadRequest(reason=msg, body=json.dumps({"error": msg}))
+
+            # Validate existing token
+            ret = cls.validate_token(bearer_token)
+            if ret.get('error') is None:
+                return web.json_response(ret)
+            else:
+                msg = ret['error']
+                raise web.HTTPBadRequest(reason=msg, body=json.dumps({"error": msg}))
+        else:
+            msg = 'bearer token parameter is missing'
+            raise web.HTTPBadRequest(reason=msg, body=json.dumps({"error": msg}))
+
+    @classmethod
+    def validate_token(cls, token):
+        """ Validate service bearer token
+        """
+        try:
+            ret = jwt.decode(token,
+                         SERVICE_JWT_SECRET,
+                         algorithms=[SERVICE_JWT_ALGORITHM],
+                         options = {"verify_signature": True, "verify_aud": False, "verify_exp": True})
+            return ret
+        except Exception as e:
+            return { 'error' : str(e) }
+
+    @classmethod
+    async def refresh_token(cls, request):
+        """ Endpoint for refresh of service bearer token received at registration time
+
+        :Example:
+            curl -X POST -d '{"bearer_token" : "hjhdjsfshhj"}
+             -H 'Authorization: ZGdrdmV4dWFsY2dsaHVyZmFxdmdybXB5dXduaXJvc3g='
+             http://localhost:<core mngmt port>/fledge/service/refresh_token
+
+        Authorization header must contain the startup token in base64 format
+
+        Note: token will be refresh for the service it belongs
+        """
+
+        data = await request.json()
+        authData = request.headers.get('Authorization', "")
+        auth = base64.b64decode(authData).decode('utf-8')
+        bearer_token = data.get('bearer_token',"")
+
+        try:
+            for k, v in cls.scheduler._startupTokens.items():
```

Comment:
why do we need to walk through startup/single use tokens for refreshing the token?
I believe we need to decode the refresh token and then use the subject to find ServiceRegistry.get(name={}) and then match the token, refresh and assign new.